### PR TITLE
KeyVault: adding support for specifying key size on creation

### PR DESCRIPTION
**Pull request by dragav**
## Description

Allowing the specification of key size on creation.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/KeyVault/Commands.KeyVault/Models/PSKeyVaultKeyAttributes.cs [hunk 1]
```diff
@@ -26,8 +26,7 @@ public class PSKeyVaultKeyAttributes
         public PSKeyVaultKeyAttributes()
         { }
 
-        internal PSKeyVaultKeyAttributes(bool? enabled, DateTime? expires, DateTime? notBefore, string keyType,
-            string[] keyOps, Hashtable tags)
+        internal PSKeyVaultKeyAttributes(bool? enabled, DateTime? expires, DateTime? notBefore, string keyType, string[] keyOps, Hashtable tags)
```
Comment (maddieclayton) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Models/PSKeyVaultKeyAttributes.cs]: Did you mean to change this include the size?

Comment (dragav) [reply src/ResourceManager/KeyVault/Commands.KeyVault/Models/PSKeyVaultKeyAttributes.cs]: No, this is intentional. The size remains strictly a creation-time parameter, and is not captured in any client-side data type. This matches the behavior of the SDK, REST API and the service, respectively.

File: src/ResourceManager/KeyVault/Commands.KeyVault/help/Remove-AzureKeyVaultManagedStorageAccount.md [hunk 2]
```diff
@@ -83,6 +95,19 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -InRemovedState
+Permanently remove the previously deleted managed storage account.```yaml
```
Comment (maddieclayton) [inline src/ResourceManager/KeyVault/Commands.KeyVault/help/Remove-AzureKeyVaultManagedStorageAccount.md]: Move this onto a new line.

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/RemoveAzureKeyVaultManagedStorageAccount.cs [hunk 3]
```diff
@@ -43,6 +44,13 @@ public class RemoveAzureKeyVaultManagedStorageAccount : KeyVaultCmdletBase
         [Alias( Constants.StorageAccountName, Constants.Name )]
         public string AccountName { get; set; }
 
+        /// <summary>
+        /// If present, operate on the deleted entity.
+        /// </summary>
+        [Parameter(Mandatory = false,
+                    HelpMessage = "Permanently remove the previously deleted managed storage account.")]
+        public SwitchParameter InRemovedState { get; set; }
```
Comment (maddieclayton) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/RemoveAzureKeyVaultManagedStorageAccount.cs]: Are you planning to do this for ManagedStorageSasDefinition as well?

File: src/ResourceManager/KeyVault/Commands.KeyVault/help/Remove-AzureKeyVaultManagedStorageAccount.md (hunk 2 above)
Comment (dragav) [reply src/ResourceManager/KeyVault/Commands.KeyVault/help/Remove-AzureKeyVaultManagedStorageAccount.md]: Thanks, fixed. Not sure why platyPs generates this on the same line.

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/RemoveAzureKeyVaultManagedStorageAccount.cs (hunk 3 above)
Comment (dragav) [reply src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/RemoveAzureKeyVaultManagedStorageAccount.cs]: No, sas definitions are considered aspects of a composite entity (the managed storage account), and thus can't be deleted/purged independently of the parent object - that is, the service itself enforces this behavior.

**Comment by maddieclayton** [comment]
on demand job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/743/

**Pull request merged**